content/authoring/kata/create-first-kata.md

```diff
@@ -0,0 +1,266 @@
+
+While the old kata are usually lazy and only have a few tests, nowadays if you try to pull off the same trick you'll instantly get yelled at with all the `Needs random tests!!!!!11!1` and your satisfaction rating plummets to 0%.
+
+The key thing about tests is that a test should perform two things:
+1. Accept all conforming solutions
+2. Reject all non-conforming solutions
+
+Some people might think that only point 1 is necessary, which is obviously untrue: test that accepts **everything** is pointless. Good tests will let the correct solutions -- *and only* the correct solutions -- pass.
+
+For normal kata, a good set of tests should cover all of these aspects:
+- Test basic functionality
+- Has full coverage (if that's impossible, at least have decent coverage)
+- Cover edge cases thoroughly
+- Randomized tests to probe user solution with random samples (and so that pattern-matching against the tests is impossible)
+- Stress/performance/code characteristic tests if needed
+
+The first three should be put into fixed tests. The fourth item should be put into random tests (see below). Ideally the last item would be in isolation, or covered by random tests. The last item is optional.
+
+Random test cases are test cases that cannot be predicted. Most kata have them (except for the really old ones) and they are usually in addition to some static tests. Using both static and random test cases with make it both easy for warriors to see what they are supposed to do, as well as make it impossible to write a solution that just pattern match the inputs (i.e return hard-coded outputs depending on a fixed set of inputs). Random tests are also good at finding edge cases.
+
+Remember: just like in real life, if we failed a test, we want to know:
+- what was the input
+- expected and actual result
+
+So unless revealing the expected result would spoil the kata, you should not hide them. Consult the documentation of testing framework used by your [language](/languages/) and pick the best method for your tests.
+
+Writing good test suites is very difficult, often the most difficult part of creatign a kata. It often requires more code than the actual solution, and any flaws in the test suite can heavily impact the overall quality and experience of users with the kata. Make sure you follow [quality guidelines for submssion tests](/authoring/guidelines/submission-tests/) when writing them, as they will help you to avoid many common pitfalls awaiting for inexperienced authors.
```

Comment:
```suggestion
Writing good test suites is very difficult, often the most difficult part of creating a kata. It often requires more code than the actual solution, and any flaws in the test suite can heavily impact the overall quality and experience of users with the kata. Make sure you follow [quality guidelines for submission tests](/authoring/guidelines/submission-tests/) when writing them, as they will help you to avoid many common pitfalls awaiting for inexperienced authors.
```